sample-app/e2e/functional.test.ts

```diff
@@ -0,0 +1,174 @@
+import {element, expect} from 'detox'
+import {Selectors, TestData, UserFeedback} from './constants'
+import {
+  fillPaymentDetailsSheet,
+  assertResultsScreen,
+  complete3DS2,
+  addCardPaymentMethod,
+  setNoPreferenceCRI,
+  clickSettingsButton,
+  enterAuthDetails,
+} from './helpers'
+
+describe('E2E Functional Tests', () => {
+  beforeAll(async () => {
+    await device.launchApp()
+    await clickSettingsButton()
+    await enterAuthDetails()
+  })
+
+  beforeEach(async () => {
+    await device.launchApp()
+  })
+
+  afterEach(async () => {
+    await device.terminateApp()
+  })
+
+  it('should successfully complete a 3DS2 token payment transaction', async () => {
+    await element(by.id(Selectors.FEATURE_LIST)).scrollTo('bottom')
+    await element(by.text(Selectors.TOKEN_PAYMENTS)).tap()
+    await element(by.text(Selectors.TOKENIZE_NEW_CARD)).tap()
+    await fillPaymentDetailsSheet()
+    await element(by.id(Selectors.TOKEN_CVV_CODE)).typeText(
+      TestData.SECURITY_CODE,
+    )
+    await element(by.id(Selectors.TOKEN_SCROLL_VIEW)).scrollTo('bottom')
+    await element(by.id(Selectors.PAY_WITH_TOKEN)).tap()
+    await complete3DS2()
+    try {
+      await element(by.text(Selectors.THREEDS2_COMPLETE_BUTTON)).longPress()
+      await element(by.text(Selectors.THREEDS2_COMPLETE_BUTTON)).longPress()
+    } catch (error) {}
+    await assertResultsScreen('1', '1')
```

Comment:
` assertResultsScreen('1', '1')` - these are magic numbers, and it's unclear what is what,
I would recommend using well-named enums or constants for this purpose and maybe having something like this is more clear:


```
export interface Props {
    type: string,
    result: string
}

export async function assertResultsScreen(props: Props) => {}


then you can invoke it like so:

await assertResultsScreen({ type: "1", result: "1" })
```

Reply:
Superb, thank you for this! I'll implement it this way

Reply:
Superb, thank you for this! I'll implement it this way